documentation/src/main/asciidoc/topics/contributing/terminology.adoc

```diff
@@ -0,0 +1,88 @@
+[[terminology]]
+= {brandname} Terminology
+
+{brandname} terminology refers to words and phrases specific to {brandname}
+capabilities and features. It is important to use consistent terminology across
+documentation and the code base to avoid unclear or contradictory language.
+
+Add::
+Add an item to an existing list, group, view, or other container element. For example, "add entries to caches".
+
+Cache configuration::
+An XML or JSON document that controls {brandname} behavior.
```

Comment:
Isn't this too general to define a cache configuration?

Reply:
@rigazilla Maybe but I'm not sure. What do you think is missing or how should we change the definition? 

For me, the "cache configuration" is the thing/artefact you use to enable features and define how you want Infinispan to behave. Which is distinct to a "configuration template" (a thing from which you create configurations). In docs I've seen configuration, template, and definition used inconsistently/interchangeably. Cheers.

Reply:
@oraNod I also think this definition is too broad. If we had a definition of "Cache" first, then the definition of "Cache configuration" could be "The specification of a cache".

OTOH it's not wide enough, because a configuration may be defined in a standalone XML/JSON document, as an element in the Infinispan embedded/server XML, or with Java code calling `ConfigurationBuilder` methods. Also we call a `Configuration` object a "cache configuration", even if it's not connected to an existing cache, or even registered in a cache manager.

In the end, I think the cache configuration/cache configuration template is a bit artificial right now, and we can't make this distinction as long as the same `DefaultCacheManager.defineConfiguration()` method is used to define both caches and templates.

Reply:
One more thing that probably needs to change for this to make sense: currently, a `Configuration` object does not include the cache name.

Perhaps that's why in some places we use "cache definition" instead of "cache configuration: Because "cache configuration" can be interpreted as the `Configuration` object, without the cache name, and XML document or element defining a cache does include the cache name, so it's somehow more than just a cache configuration.

/cc @tristantarrant @ryanemerson

Reply:
thanks for all those details @danberindei I need more time to digest that before making an update. IMO it's really good to start having this conversation to tackle some complexity. what you're saying helps point out the fact that what seems like a straightforward terminology definition isn't all that clear cut, which I think can come across to users.

Reply:
+1, XML or JSON is just about the format

Reply:
> @rigazilla Maybe but I'm not sure. What do you think is missing or how should we change the definition?
> 
> For me, the "cache configuration" is the thing/artefact you use to enable features and define how you want Infinispan to behave. Which is distinct to a "configuration template" (a thing from which you create configurations). In docs I've seen configuration, template, and definition used inconsistently/interchangeably. Cheers.

@oraNod forgive my delay, I've lost myself on another front.
I would add to the `cache definition` some more details to narrow down the part of Infinispan that a `cache definition` controls: i.e. we're going to control/represent a cache resource.
This is what it seems to me that is missing reading the definition.

Anyway @danberindei did a more detailed analysis above :)

Reply:
@danberindei and @rigazilla thanks, guys. I've updated this part. Honestly I'm of two minds about including terms for "cache configuration", "cache template", and "cache definition". Perhaps it doesn't make that much sense to create narrow definitions for these terms and we should remove them. 

The whole intent of this PR is to have general guidelines for improving consistency within UI (console, CLI, log messages) and documentation. Maybe I'm inventing an issue where it doesn't really exist but I feel like we should be able to tell users "when we say 'cache configuration' we mean x" instead of "we mean x but in some cases, we also mean y".

Reply:
> @danberindei and @rigazilla thanks, guys. I've updated this part. Honestly I'm of two minds about including terms for "cache configuration", "cache template", and "cache definition". Perhaps it doesn't make that much sense to create narrow definitions for these terms and we should remove them.

@oraNod the new version looks better from my POV, IMO pragmatical guidelines are good enough for a contributing guide, if a sharp definition make things harder